Automattic/WP-Job-Manager
# Add date field for frontend submit form

This gives extensions the ability to add date fields to the job posting form on the frontend while taking advantage of WPJM's datepicker and date handling.

## Testing

- Install and activate Application Deadline and use the code from this PR: https://github.com/Automattic/wp-job-manager-application-deadline/pull/22

- Ensure that the date pickers on the Edit Job page in WP Admin work properly.

- Go to the frontend page to post a job (i.e. the page that uses the `submit_job_form` shortcode). The "Closing date" field should have a datepicker. When the job is submitted, that date should be set correctly.

- Test the datepickers when using different locales. Ensure that they are translated properly (they should all use the datepicker localization from WordPress) and that choosing different date formats does not break the functionality (see https://github.com/Automattic/wp-job-manager-application-deadline/issues/8)

- When Application Deadline is disabled, the jQuery datepicker library should not be loaded on the frontend. Verify this by ensuring that `jQuery.fn.datepicker` is undefined.
